### ISSUE TITLE
Forbid Word-ish items

### ISSUE THREAD

**Issue by scout249** [check-spelling/check-spelling#18]
I have setup **reject.txt** and **allow.txt** as below. I want to identify the words in reject.txt as misspelling, no matter these words are in the dictionary or not.

How do I know if the reject list is being used or not? It seem to me nothing have happened after all.

**Reject.txt**
e-share
e-Share
E-Share
E-share
eShare
eshare
Eshare

**allow.txt**
EShare

**Comment by jsoref** [comment]
Sadly at this time there's no way to do anything in the `-` family from the . I have a plan for it, but it isn't currently supported. Currently, the tool sees all of those as two words `e`/`E` (which it ignores since it only checks words with at least 3 letters by default) and `share`/`Share`.

Also note that `eShare` will be parsed as `e` `Share` -- and thus is equivalent to `e-Share` in that the only word seen is `Share` which you aren't removing from the dictionary.

`reject` removes things from the base dictionary.
Since I doubt any of the items you've listed in `reject.txt` are in the dictionary, I expect that it indeed won't do anything today.

`allows` adds things to the base dictionary. Thus, I'd expect `EShare` to work.

## Today

For now, you could *cheat* by doing this:

`reject.txt`:
```
share
Share
```

`allow.txt`:
```
EShare
```

`patterns.txt`:
```
(?:^|\s)(?i)share\b
```

## Coming

To eventually handle your use case, I intend to add something like: [Forbidden patterns](https://github.com/check-spelling/check-spelling/wiki/Feature%3A-Forbidden-patterns)

With that, you'd add:
`forbidden.txt`:
```
\b(?i)e-share\b
\beShare\b
\beshare\b
\bEshare\b
```

I've only recently started thinking about how to do this, I know roughly where to put the code and am still thinking about the messaging for it within the output. It'll be a _bit_ different, since I know that some people will use it not just to reject words, but actual coding patterns (the reason I started thinking about it was that someone wanted to forbid `class Something` -- I can't remember the precise example, but it was clearly not a "word").

**Comment by jsoref** [comment]
I have a first draft of this feature:
https://github.com/jsoref/check-spelling/commit/be18f271ffeb2dd0bc809c375b7589312ee5de14

It's sitting at the top of my feature queue, unfortunately, in doing so, I've added some other things that I've been stalled on. I think it's a pretty good change, but it means I'm going to want a bit more baking time.

**Comment by jsoref** [comment]
This shipped in [v0.0.20](https://github.com/check-spelling/check-spelling/releases/tag/v0.0.20) as [Forbidden patterns](https://github.com/check-spelling/check-spelling/wiki/Feature:-Forbidden-patterns)
